example/conv2DWithMdspan/src/conv2DWithMdspan.cpp

```diff
@@ -0,0 +1,243 @@
+/* Copyright 2023 Mehmet Yusufoglu, Bernhard Manfred Gruber, René Widera
+ * SPDX-License-Identifier: ISC
+ */
+
+#include <alpaka/alpaka.hpp>
+// Needed for running example for all backends available; one by one
+#include <alpaka/example/ExecuteForEachAccTag.hpp>
+
+#include <iomanip>
+#include <iostream>
+#include <vector>
+
+//! Convolution Example using Mdspan structure to pass multi-dimensional data to the kernel
+//!
+//! A 2D Convolutional filter applied to a matrix. Mdspan data structure is used, therefore pitch and size values are
+//! not needed to be passed to the kernel. Results can be tested by comparing with the results of the Matlab call: Y =
+//! filter2(FilterMatrix,InputMatrix,'same');
+
+/**
+ * @brief 2D Convolutional Filter using only global memory for the input-matrix and the filter-matrix
+ */
+struct ConvolutionKernelMdspan2D
+{
+    //! \tparam TAcc Accelerator type
+    //! \tparam TElem The input-matrix and filter-matrix element type
+    //! \param acc Accelerator
+    //! \param input Input matrix
+    //! \param output Output matrix
+    //! \param filter Filter-matrix
+
+    template<typename TAcc, typename MdSpan>
+    ALPAKA_FN_ACC auto operator()(TAcc const& acc, MdSpan const input, MdSpan output, MdSpan const filter) const
+        -> void
+    {
+        auto matrixWidth = input.extent(0);
+        auto matrixHeight = input.extent(1);
+        // Filter matrix is square
+        auto filterWidth = filter.extent(0);
```

Comment:
```suggestion
        auto filterWidth = static_cast<int32_t>(filter.extent(0));
```
and remove the `static_cast` below ?